drivers: flash: shell: add verify step to cmd_test

heinwessels opened the pull request:
Previously cmd_test could falsely give the indication that the flash driver functioning properly, because the written data isn't validated, and thus not truley tested.

This commit a verify step to ensure the data was written successfully.

Example output:
```
uart:~$ flash test flash-controller@52002000 100000 1000
Erase OK.
Write OK.
Verified OK.
Erase-Write-Verify test done.
```

```
uart:~$ flash test qspi-nor-flash@0 0 1000
Erase OK.
Write OK.
Verification ERROR!
Erase-Write-Verify test done.
```

Signed-off-by: Hein Wessels <heinwessels93@gmail.com>

heinwessels committed:
drivers: flash: shell: add verify step to cmd_test

Previously cmd_test could falsely give the indication that the
flash driver is functioning properly, because the written data
isn't validated and could have silently failed.

This commit adds a verify step to ensure the data was written successfully.

Signed-off-by: Hein Wessels <heinwessels93@gmail.com>

nordicjm commented on drivers/flash/flash_shell.c:
```diff
@@ -241,11 +247,23 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 
 		shell_print(shell, "Write OK.");
+
+		if (flash_read(flash_dev, addr, check_arr, size) < 0) {
+			shell_print(shell, "Verification read ERROR!");
```
Why not use result and output error as above?

nordicjm commented on drivers/flash/flash_shell.c:
```diff
@@ -241,11 +247,23 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
+			break;
+		}
+
+		shell_print(shell, "Verified OK.");
```
The part below should be updated to be "Erase-Write-Verify test done."

heinwessels commented:
@nashif @de-nordic Pinging here, because requesting re-review dismisses all other pending reviews.

de-nordic commented on drivers/flash/flash_shell.c:
```diff
@@ -222,6 +224,10 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 		return -EINVAL;
 	}
 
+	if (repeat == 0) {
+		repeat = 1;
+	}
+
```
This should have separate commit as it does not fix what the commit describes.

heinwessels replied:
It's now in a separate commit with explanation.

heinwessels committed:
drivers: flash: shell: default test repeat parameter when not supplied

Previously when the repeat argument wasn't supplied to the shell
command it would falsely seem to succeed, even though the
test was never run. Instead of displaying an error message this
rather overrides the repeat to 1, so that the test is run once.

Signed-off-by: Hein Wessels <heinwessels93@gmail.com>

heinwessels commented:
> @heinwessels In the stabilization phase, we need a bug linked to each PR. If you'd like to merge before release, file a bug and let me know +1

@laurenmurphyx64 this is not blocking, and I don't mind waiting for the next development phase.
